### ISSUE TITLE
OSX: Requirement to install libffi is never stated

### ISSUE THREAD

**Issue by stopspazzing** [certbot/certbot#1970]
While trying to run letsencrypt-auto on OSX after installing homebrew, I was getting a failing program-- a mess of maybe 10 lines of error messages.

Unlike a configure script, letsencrypt-auto did not help me solve the problem of letsencrypt-auto failing in a mess.

This is just a note to whoever is developing letsencrypt-auto: It is easy to assume that this is completely broken software because it's not stated clearly that libffi is vitally required. It is easy to become completely distracted by the error messages and try to solve the wrong problem to get this running. That wastes people's time and makes them want to give up. If you put out software that wastes people's time, even if they're just spending their spare time which is worth $0/hour, it leaves a bad taste and discourages adoption.

Put this in the docs please:

Only after installing libffi will a curses UI even appear.

This is a quote from forums: https://community.letsencrypt.org/t/requirement-to-install-libffi-is-never-stated/7475

This is similar to my issue: https://github.com/letsencrypt/letsencrypt/issues/1888

**Comment by reaperhulk** [comment]
The reason libffi is not listed as an installation requirement is because libffi, while required, ships with OS X. In this particular case you almost certainly ran into a bug with the cffi 1.4.1 wheels shipped on PyPI that were improperly linked against homebrew libffi rather than the system libffi. When this bug was discovered the wheels were deleted, but any user with the cached wheel would continue to have the problem. This was subsequently resolved with the release of cffi 1.4.2. This issue was an unfortunate confluence of several complex interactions and was not the fault of lets encrypt in any way.

I realize this is a copy from the forums, but I have to say that it's unfortunate that the author decided to imply that the hard work of so many people is "wasting people's time". I write this on issues frequently, but I urge everyone to remember that even bugs that appear simple (like "you forgot to document a libffi requirement") may have a far more complex explanation.

**Comment by sydneyli** [comment]
It sounds like this was an upstream issue; also, we no longer use curses as of #3324, so this is probably no longer relevant. Feel free to bug me if I'm wrong about this and I'll be happy to re-open.
